app/queries/subscriber_list_query.rb

```diff
@@ -71,12 +76,14 @@ def subscriber_lists_with_keys_overlapping(query_hash)
   # then this returns all lists which have any "topics" AND "organisations"
   # links.
   def subscriber_lists_with_all_matching_keys(query_hash)
-    # This uses the `?&` hstore operator, which returns true only if the hstore
+    # This uses array equality to check if the JSON object
     # contains all the specified keys.
-    SubscriberList.where("#{@query_field} ?& Array[:keys]", keys: query_hash.keys)
+    SubscriberList.where("ARRAY(SELECT json_object_keys(#{@query_field})) = Array[:keys]",
+      keys: query_hash.keys,
+    )
   end
 
   def subscriber_lists_without_tags_or_links
-    SubscriberList.where(tags: "", links: "")
+    SubscriberList.where("tags_json::text = '{}'::text AND links_json::text = '{}'::text")
```

Comment:
I don't know if it will make any real difference, but you should also be able to write
`tags_json = '{}'::json AND links_json = '{}'::json` to skip some of that casting.